tools/autograd/templates/Functions.cpp

```diff
@@ -809,10 +799,545 @@ Tensor softplus_double_backward(const Tensor & grad, const Tensor & input, Scala
   return _sigmoid_backward(grad, x.sigmoid()) * (x < threshold).toType(grad.type()) * beta;
 }
 
-Tensor as_strided_backward(const Tensor & grad, TensorGeometry base, IntList sizes, IntList strides, int64_t storage_offset) {
-  auto src = base.zeros_with_stride(grad.type());
-  src.as_strided(sizes, strides, storage_offset - base.storage_offset()).copy_(grad);
-  return src;
+
+// NOTE [ as_strided Backward ]
+//
+// `storage_offset` is ignored for simplicity in this note. If you just want the
```

Comment:
This is really long. Let's move this to a separate file.

I think we should pull in `mem_overlap.c` from NumPy instead of trying to re-solve this ourselves. It contains code that solves this problem precisely or with a bound on the amount of work. It doesn't depend to heavily on the NumPy API. Most of the porting will be mechanical.